From c0c84f0619043e833aabfd96b5c43b35fbff2f5d Mon Sep 17 00:00:00 2001
From: zlong2 <zhifang.long@intel.com>
Date: Fri, 13 Apr 2018 09:18:08 +0800
Subject: [PATCH 3/7] multidroid : support multi instances on multi screens

n instances on n screens :
instance 1 will be displayed on screen 2
instance x(<n) will be displayed on screen x+1
instance n will be displayed on screen 1

Change-Id: I17fb316418b3213d293ab288e817bf753bd4cdd5
Tracked-On:
Signed-off-by: zlong2 <zhifang.long@intel.com>
Reviewed-on: 625937
Reviewed-by: Zhang, Hongyu <hongyu.zhang@intel.com>
Tested-by: Zhang, Hongyu <hongyu.zhang@intel.com>
Reviewed-on: 674560
Tested-by: Xue, Yifei <yifei.xue@intel.com>
Reviewed-by: Shi, Zhongjie <zhongjie.shi@intel.com>
---
 drv.c | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/drv.c b/drv.c
index aa797ce..0f68fb7 100755
--- a/drv.c
+++ b/drv.c
@@ -1123,18 +1123,17 @@ int drv_init_kms(struct driver* drv)
             ALOGW("cxt.process_devns_id for %d\n", cxt.process_devns_id);
             kms->lxc_id = cxt.process_devns_id;
         }
-        //    char lxc_prop[PROPERTY_VALUE_MAX];    
-        //    memset(lxc_prop, 0, sizeof(lxc_prop));
-        //    property_get("lxc.instance.no", lxc_prop, "1");
-        //    lxc_id = atoi(lxc_prop);    
     }
 #else
-    kms->lxc_id = 1;
+    {
+        char prop_id[PROPERTY_VALUE_MAX];
+        memset(prop_id, 0, sizeof(prop_id));
+        property_get("sys.container.id", prop_id, "0");
+        kms->lxc_id  = atoi(prop_id) + 1;
+    }
 #endif
     ALOGD("LXC id = %d", kms->lxc_id);    
 
-
-
     {
         uint32_t connector_id;
         uint32_t encoder_id;
@@ -1161,7 +1160,7 @@ int drv_init_kms(struct driver* drv)
     }
 
     if(connected_count > 1) {
-        kms->lxc_id += LXC_SHIFT_DISPLAY;
+        kms->lxc_id = kms->lxc_id % connected_count + LXC_SHIFT_DISPLAY;
         ALOGD("LXC id = %d after shift", kms->lxc_id);
     }
 
-- 
2.21.0

